src/main/java/com/linkedin/kafka/clients/producer/LiKafkaProducerImpl.java

```diff
@@ -342,4 +399,14 @@ public void onCompletion(RecordMetadata recordMetadata, Exception e) {
     }
   }
 
+  private final class SimplePartitionerImpl implements SimplePartitioner {
```

Comment:
I am actually not sure if this is necessary. Usually we want to avoid having something very similar to the existing configurations. Ideally we want to just use the open source partitioner. I understand that we do not have easy access to the Cluster which is required by the open source Kafka partitioner interface, maybe we can pass in null in this case. If there is no good solution, I would rather remove this partitioner at this point. It is always much easier to add something than remove some existing things.